lib/Driver/Driver.cpp

```diff
@@ -827,25 +841,49 @@ Driver::buildCompilation(const ToolChain &TC,
     else
       llvm_unreachable("Unknown OutputLevel argument!");
   }
-
-  std::unique_ptr<Compilation> C(
-      new Compilation(Diags, TC, OI, Level,
-                      std::move(ArgList),
-                      std::move(TranslatedArgList),
-                      std::move(Inputs),
-                      buildRecordPath,
-                      outputBuildRecordForModuleOnlyBuild,
-                      ArgsHash,
-                      StartTime,
-                      LastBuildTime,
-                      DriverFilelistThreshold,
-                      Incremental,
-                      BatchMode,
-                      DriverBatchSeed,
-                      DriverForceOneBatchRepartition,
-                      SaveTemps,
-                      ShowDriverTimeCompilation,
-                      std::move(StatsReporter)));
+  
+  // About to move argument list, so capture some flags which will be needed
+  // later.
+  const bool DriverPrintActions =
+      ArgList->hasArg(options::OPT_driver_print_actions);
+  const bool DriverPrintDerivedOutputFileMap =
+      ArgList->hasArg(options::OPT_driver_print_derived_output_file_map);
+  const bool ContinueBuildingAfterErrors =
+      computeContinueBuildingAfterErrors(BatchMode, ArgList.get());
+  const bool ShowJobLifecycle =
+      ArgList->hasArg(options::OPT_driver_show_job_lifecycle);
+
+  // In order to confine the values below, while still moving the argument
+  // list, and preserving the interface to Compilation, enclose the call to the
+  // constructor in a block:
+  std::unique_ptr<Compilation> C;
+  {
+    const bool SaveTemps = ArgList->hasArg(options::OPT_save_temps);
+    const bool ShowDriverTimeCompilation =
+        ArgList->hasArg(options::OPT_driver_time_compilation);
+    std::unique_ptr<UnifiedStatsReporter> StatsReporter =
+        computeStatsReporter(ArgList.get(), Inputs, OI, DefaultTargetTriple);
+    
+    C =
+    llvm::make_unique<Compilation>(
```

Comment:
Nit: `clang-format` wants this to be 

```
    C = llvm::make_unique<Compilation>(
        Diags, TC, OI, Level, std::move(ArgList), std::move(TranslatedArgList),
        std::move(Inputs), buildRecordPath, outputBuildRecordForModuleOnlyBuild,
        ArgsHash, StartTime, LastBuildTime, DriverFilelistThreshold,
        Incremental, BatchMode, DriverBatchSeed, DriverForceOneBatchRepartition,
        SaveTemps, ShowDriverTimeCompilation, std::move(StatsReporter));
```

but I like the stacked argument list here, so I guess an outdent of the argument list would split the difference.

Reply:
Yes, I'll outdent it. This is a case where git-clang-format does the "wrong" thing and makes the deltas harder to read.